File: backend/SmartSusChef.Api/Models/Ingredient.cs
namespace SmartSusChef.Api.Models;

public class Ingredient
{
    public Guid Id { get; set; }
    // Foreign Key for the Store
    public int StoreId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Unit { get; set; } = string.Empty;
    public decimal CarbonFootprint { get; set; } // kg CO2 per unit
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    // Link to the Store entity
    public Store Store { get; set; } = null!;
    public ICollection<RecipeIngredient> RecipeIngredients { get; set; } = new List<RecipeIngredient>();
    public ICollection<WastageData> WastageRecords { get; set; } = new List<WastageData>();
}

File: mobile/app/src/main/java/com/smartsuschef/mobile/data/models/Ingredient.kt
package com.smartsuschef.mobile.data.models

import java.util.Date

data class Ingredient(
    val id: String, // Guid (.NET) --> String in Kotlin
    val name: String,
    val storeId: Int,
    val unit: String,
    val carbonFootprint: Double, // decimal (.NET) --> double in Kotlin
    val createdAt: String, // DateTime (.NET) -> String (ISO format) in Kotlin
    val updatedAt: String
)
